components/automate-ui/src/app/components/chart-progress-bar/chart-progress-bar.component.ts

```diff
@@ -0,0 +1,123 @@
+import { Component, Input, OnInit, OnChanges } from '@angular/core';
+import { Styles } from './chart-progress-bar.model';
+
+@Component({
+  selector: 'app-chart-progress-bar',
+  templateUrl: './chart-progress-bar.component.html',
+  styleUrls: [ './chart-progress-bar.component.scss' ]
+})
+
+export class ChartProgressBarComponent implements OnInit, OnChanges {
+
+  @Input() size = 'large'; // large or small
+  @Input() value: number;
+  @Input() min: number;
+  @Input() max: number;
+  @Input() threshold: number[]; // [min, max]
+  @Input() valueTooltip: any;
+  @Input() maxTooltip: any;
+
+  public styles: Styles = {
+    bar: {},
+    container: {},
+    progress: {},
+    threshold: { point1: {}, point2: {}}
+  };
+  public tooltips = {
+    value: '',
+    max: ''
+  };
+
+  constructor() {}
+
+  ngOnInit() {
+   this.createStackedBar();
+   this.populateStackedBar();
+  }
+
+  ngOnChanges() {
+    this.populateStackedBar();
+  }
+
+  public createStackedBar() {
+    this.size === 'large' ? this.createLargeBar() : this.createSmallBar();
+  }
+
+  public populateStackedBar() {
+    this.populateValueNow();
+    this.populateTooltips();
+    this.populateThreshold();
+    this.populateProgressBackground();
+  }
+
+  public createLargeBar() {
+    this.styles.bar = { borderRadius: '8px', height: '16px' };
+    this.styles.container.height = '16px';
+    this.styles.progress.borderRadius = '20px';
+    this.styles.progress.height = '16px';
+    this.styles.threshold.height = '16px';
+  }
+
+  public createSmallBar() {
+    this.styles.bar.background = 'none';
+    this.styles.bar.boxShadow = 'none';
+    this.styles.progress.borderRadius = '8px';
+    this.styles.container.height = '8px';
+    this.styles.progress.height = '8px';
+    this.styles.threshold.height = '8px';
+  }
+
+  public populateValueNow() {
+    const value = this.getValuePercentage(this.value);
+    this.styles.bar.width = (100 - value) + '%';
+    this.styles.progress.width = value + '%';
+
+    // Have no progress bar right radius
+    // until progress reaches 100%
+    this.styles.progress.borderRadius =
+      (this.size === 'large' && value < 99)
+      ? '20px 0 0 20px'
+      : '20px';
+
+    // Have no bar left radius
+    // when progress is greater than 1%
+    this.styles.bar.borderRadius =
+    (this.size === 'large' && value > 0)
+    ? '0 20px 20px 0'
+    : '20px';
```

Comment:
an option to consider...
I was playing with this in the view, if you instead lay out a container with progress inside of it, you can set the container to overflow: hidden, border-radius 20% and then you never have to deal with calculating when to use a radius, and can delete this whole area.

```
<div class="container">
  <div id="progress" class="progress" [ngStyle]="styles.progress"></div>
</div>
```

You could then delete the "bar" div and also pull any functions or styles related to bar.

Reply:
@SEAjamieD it's a good recommendation. We also have to consider having tooltips for both bars that are centered based on the width of the bar that displays over hover.

Reply:
@SEAjamieD border-radius refactoring has moved into css file. 👍🏻